api/v1alpha1/api.go

```diff
@@ -378,3 +426,34 @@ type AWSOIDCExchangeToken struct {
 	// which maps to the temporary AWS security credentials exchanged using the authentication token issued by OIDC provider.
 	AwsRoleArn string `json:"awsRoleArn"`
 }
+
+// RequestCost configures the request cost.
+type RequestCost struct {
+	// Type specifies the type of the request cost. The default is "default",
+	// and it uses "output token" as the cost.
+	//
+	// +kubebuilder:validation:Enum=default
+	Type RequestCostType `json:"type"`
+
+	// CELExpression is the CEL expression to calculate the cost of the request.
+	// The CEL expression must return an integer value. The CEL expression should be
+	// able to access the request headers, model name, backend name, input/output tokens etc.
+	//
+	// +optional
+	// +notImplementedHide https://github.com/envoyproxy/ai-gateway/issues/97
+	CELExpression *string `json:"celExpression"`
```

Comment:
@yuzisun  I think this should be enough for  https://github.com/envoyproxy/ai-gateway/issues/97? also I will add some extensibility to extproc code so that users can insert their own custom extproc logic (like custom routing etc) - that's where also the custom weight calculation can be done.

Reply:
Are we going to validate cel expression in validation webhook ?

Reply:
yeah if it's possible - not sure if we can do the cel expr validation inside the cel validation rule of k8s...